### PR TITLE
Errors and ipc enums & Function prototypes

### PR DESCRIPTION
**Pull request by heyimnowi**
## Summary
- connection types enum
- request states enum
- connection errors enum
- read and write prototypes
## Doubts

Please check my comments

### CODE REVIEW

**Comment by julianrod94** [comment]
pipi cucu :+1:

File: Api/request.c [hunk 1]
```diff
@@ -1,13 +1,41 @@
 #include "ipc.h"
 #include "request.h"
 
-void myRequest() {
+// donde van todos los enums y tydefs?
```
Comment (jjsebasv) [inline Api/request.c:4]: We could do an Enums.h, couldn't we?

**Comment by jjsebasv** [comment]
:+1: 

Is this a rebase of https://github.com/jjsebasv/SO---Os/pull/5 ?

**Comment by julianrod94** [comment]
La 5 no se mergeo todavía

**Comment by julianrod94** [comment]
Es continuación de la 5

**Comment by heyimnowi** [comment]
@jjsebasv @julianrod94 First, we have to merge https://github.com/jjsebasv/SO---Os/pull/5

**Pull request merged**